src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java

```diff
@@ -780,4 +797,76 @@ public void enhanceCommitsForRepository() {
 			.forEach(asyncEventBus::post);
 	}
 
+
+	/**
+	 * Used for showing multiple graphs
+	 * @return a list with all commits made
+	 */
+	private List<Commit> getAllCommits() {
+
+		// get all commits
+		RepositoryEntity repositoryEntity = getRepositoryEntity();
+		RepositoryApi repositoryApi = repositoriesApi.getRepository(repositoryEntity.getRepositoryName());
+		Map<String, Object> parameters = getBaseParameters();
+		parameters.put("repository", repositoryApi.getRepositoryModel());
+		BranchApi branchApi = repositoryApi.getBranch("master");
+		BranchModel branch = branchApi.get();
+		CommitSubList commits = branchApi.retrieveCommitsInBranch();
+		Collection<String> commitIds = getCommitIds(commits);
+		List<Commit> commitEntities = commitIds.stream()
+			.map(commitId -> this.commits.ensureExists(repositoryEntity, commitId))
+			.collect(Collectors.toList());
+
+		return commitEntities;
+	}
+
+	@GET
+	@Path("magical-chart-data")
+	@Produces(MediaType.APPLICATION_JSON)
+
+	public List<List<Object>> getMagicalChartData() {
+
+		List<Commit> commitEntities = getAllCommits();
+
+		return getGoogleChartDataForCommits(commitEntities);
+	}
+
+	private static List<List<Object>> getGoogleChartDataForCommits(List<Commit> commitEntities) {
```

Comment:
It doesn't call Google but it represents a data format specifically for Google Charts. So the naming is fine.